xarray/core/dataset.py

```diff
@@ -3592,6 +3593,13 @@ def real(self):
     def imag(self):
         return self._unary_op(lambda x: x.imag, keep_attrs=True)(self)
 
+    def scatter(self,  x=None, y=None, hue=None, col=None, row=None,
```

Comment:
There has been talk about a dataset plotting API (e.g. #470, #1387). So perhaps the API should be `ds.plot.scatter`? with the default `ds.plot()` = `ds.plot.scatter()` for now?

For example, in the future, it'd be good to have `ds.plot.hist()` to visualize joint distributions.

Reply:
How would you suggest to refactor that? Like the trick in `DataArray.plot`, which creates `_PlotMethods`? If so, does it make sense to name all internal functions `dataset_scatter`, `dataset_hist` and so (in order not to clash with other plotting functions for `DataArray` that are in `plot.py`)? 
A different alternative is to make a new file for `Dataset` plotting functions, or even 3 files -- for dataset, dataArray, and code which is common for both (`_infer_data`...) or something like that. I'm new to this so please advise me how to proceed...

Reply:
> A different alternative is to make a new file for Dataset plotting functions, or even 3 files -- for dataset, dataArray, and code which is common for both 

I think this would be cleanest but we should defer to @shoyer / @jhamman / @fmaussion

Reply:
I like the idea of writing a separate module. Separate modules for clearly separated functionality are good -- they make things easier to understand.

>> Like the trick in DataArray.plot, which creates _PlotMethods?
> I find this bit quite confusing in the code; but it seems like the way to go...

Please do not feel committed to this current design. I agree that it is confusing, and I would probably not recommend writing things this way if we were starting over again from scratch.

Reply:
For now, I like the idea of starting with `Dataset.plot.scatter()`. We can add other Dataset plotting methods later.